src/integrations/admin/indexing-notification-integration.php

```diff
@@ -0,0 +1,245 @@
+	 *
+	 * @var Yoast_Notification_Center
+	 */
+	private $notification_center;
+
+	/**
+	 * The options helper.
+	 *
+	 * @var Options_Helper
+	 */
+	private $options_helper;
+
+	/**
+	 * The product helper.
+	 *
+	 * @var Product_Helper
+	 */
+	private $product_helper;
+
+	/**
+	 * Prominent_Words_Notifier constructor.
+	 *
+	 * @param Indexing_Integration      $indexing_integration The indexing integration.
+	 * @param Yoast_Notification_Center $notification_center  The notification center.
+	 * @param Options_Helper            $options_helper       The options helper.
+	 * @param Product_Helper            $product_helper       The product helper.
+	 */
+	public function __construct(
+		Indexing_Integration $indexing_integration,
+		Yoast_Notification_Center $notification_center,
+		Options_Helper $options_helper,
+		Product_Helper $product_helper
+	) {
+		$this->indexing_integration = $indexing_integration;
+		$this->notification_center  = $notification_center;
+		$this->options_helper       = $options_helper;
+		$this->product_helper       = $product_helper;
+	}
+
+	/**
+	 * Initializes the integration.
+	 *
+	 * Adds hooks and jobs to cleanup or add the notification when necessary.
+	 *
+	 * @return void
+	 */
+	public function register_hooks() {
+		if ( \filter_input( INPUT_GET, 'page' ) === 'wpseo_dashboard' ) {
+			\add_action( 'admin_init', [ $this, 'cleanup_notification' ] );
+		}
+
+		if ( ! \wp_next_scheduled( self::NOTIFICATION_ID ) ) {
+			\wp_schedule_event( time(), 'daily', self::NOTIFICATION_ID );
+		}
+
+		\add_action( self::NOTIFICATION_ID, [ $this, 'create_notification' ] );
+	}
+
+	/**
+	 * Returns the conditionals based in which this loadable should be active.
+	 *
+	 * @return array
+	 */
+	public static function get_conditionals() {
+		return [
+			Admin_Conditional::class,
+		];
+	}
+
+	/**
+	 * Checks whether the notification should be shown and removes
+	 * it from the notification center if this is the case.
+	 */
+	public function create_notification() {
+		$this->notification_center->add_notification( $this->notification() );
+		$notification = $this->notification_center->get_notification_by_id( self::NOTIFICATION_ID );
+
+		if ( $notification || ! $this->should_show_notification() ) {
+			return;
+		}
+
+		$this->notification_center->add_notification( $this->notification() );
+	}
+
+	/**
+	 * Checks whether the notification should not be shown anymore and removes
+	 * it from the notification center if this is the case.
+	 */
+	public function cleanup_notification() {
+		$notification = $this->notification_center->get_notification_by_id( self::NOTIFICATION_ID );
+
+		if ( $notification === null || $this->should_show_notification() ) {
+			return;
+		}
+
+		$this->notification_center->remove_notification( $notification );
+	}
+
+	/**
+	 * Checks whether the notification should be shown.
+	 *
+	 * @return bool If the notification should be shown.
+	 */
+	protected function should_show_notification() {
```

Comment:
I have a hard time understanding this method. You start by calling `is_indexation_warning_hidden` (that is clear). But later, you retrieve many of the same options from the database that you already evaluated in `is_indexation_warning_hidden`.

I also don't really understand the return statement at the end.

Can we talk about this?

Reply:
You were right, the method's logic was a mess.
I reworked it and supplied some comments to describe what happens.

Reply:
Nice work. I did ask a follow-up question :)